Subscriber/CacheKeyExtender.php

```diff
@@ -47,23 +47,23 @@ public static function getSubscribedEvents()
     /**
      * @return string
      */
-    public function onCreateCacheHash()
+    public function onCreateCacheHash(\Enlight_Event_EventArgs $args)
     {
-        $priceGroup = 0;
+        $originalHash = $args->getReturn();
 
         if (!$this->dependencyProvider->hasShop()) {
-            return json_encode(['swag_user_price_group' => $priceGroup]);
+            return null;
         }
 
         $session = $this->dependencyProvider->getSession();
         $userId = (int)$session->get('sUserId', 0);
-        $result = $this->getCustomerPriceGroupId($userId);
+        $priceGroup = $this->getCustomerPriceGroupId($userId);
 
-        if ($result !== 0) {
-            $priceGroup = $result;
+        if ($priceGroup == 0) {
```

Comment:
please use type safe comparison `===`
to ensure the type, you could cast the customer price group id

Reply:
Hey @mitelg 
sure :-)